docs/source/cpp/datatypes.rst

```diff
@@ -66,3 +66,138 @@ To instantiate data types, it is recommended to call the provided
    type = arrow::timestamp(arrow::TimeUnit::MICRO);
    // A list type of single-precision floating-point values
    type = arrow::list(arrow::float32());
+
+
+
+Type Traits
+-----------
+
+Writing code that can handle concrete :class:`arrow::DataType` subclasses would 
+be verbose, if it weren't for type traits. They are empty structs with type 
```

Comment:
Note that type traits are a general C++ concept, it may be worth linking this concept to `<type_traits>`

Reply:
Thanks! That's helpful